recipes/opentdf-client/all/conanfile.py

```diff
@@ -36,63 +39,105 @@ def _minimum_cpp_standard(self):
+            check_min_cppstd(self, self._minimum_cpp_standard)
         min_version = self._minimum_compilers_version.get(str(self.settings.compiler))
         if not min_version:
-            self.output.warn("{} recipe lacks information about the {} compiler support.".format(
-                self.name, self.settings.compiler))
+            self.output.warn(f"{self.name} recipe lacks information about the {self.settings.compiler} compiler support.")
         else:
-            if tools.Version(self.settings.compiler.version) < min_version:
-                raise ConanInvalidConfiguration("{} requires C++{} support. The current compiler {} {} does not support it.".format(
-                    self.name, self._minimum_cpp_standard, self.settings.compiler, self.settings.compiler.version))
+            if Version(self.settings.compiler.version) < min_version:
+                raise ConanInvalidConfiguration(f"{self.name} requires C++{self._minimum_cpp_standard} support."
+                                                 "The current compiler {self.settings.compiler} {self.settings.compiler.version} does not support it.")
 
     def requirements(self):
-        self.requires("openssl/1.1.1l@")
-        self.requires("boost/1.76.0@")
-        self.requires("zlib/1.2.11@")
-        self.requires("ms-gsl/2.1.0@")
-        self.requires("libxml2/2.9.10@")
-        self.requires("libarchive/3.5.1@")
-        self.requires("nlohmann_json/3.10.4@")
-        self.requires("jwt-cpp/0.4.0@")
+        self.requires("openssl/1.1.1q")
+        self.requires("boost/1.79.0")
+        self.requires("ms-gsl/2.1.0")
+        self.requires("libxml2/2.9.14")
+        self.requires("nlohmann_json/3.11.1")
+        self.requires("jwt-cpp/0.4.0")
+        if self.options.get_safe("with_libarchive"):
+            self.requires("libarchive/3.6.1")
+        # We do not require libiconv but conan-center only allows 'stock' references, and boost+libxml2
+        # specify differerent versions, which causes a build fail due to the dependency conflict.
+        # Overriding the version here allows a clean build with the stock build settings.
+        # Preferred solution is to set with_libiconv to False
+        if self.options.get_safe("with_libiconv"):
+            self.requires("libiconv/1.17@")
+        # Same for zlib
+        # Preferred solution is to set with_zlib to False
+        if self.options.get_safe("with_zlib"):
+            self.requires("zlib/1.2.12@")
 
     def config_options(self):
+        # Do not require libarchive as of 1.1.0
+        if Version(self.version) < "1.1.0":
+            self.options.with_libarchive = True
         if self.settings.os == "Windows":
             del self.options.fPIC
+        if not self.options.with_zlib:
+            self.options["libxml2"].zlib = False
+        if not self.options.with_libiconv:
+            self.options["boost"].without_locale = True
+            self.options["boost"].without_log = True
```

Comment:
```suggestion
```
Boost 1.79 now consume latest libiconv and zlib versions

Reply:
True, but a local build can still specify those options to create smaller libs

Reply:
What do you mean, opentdf-client depends on zlib and libiconv directly?

Reply:
Not directly, or intentionally, no.  My concern is that boost and libxml2 may indirectly pull in pieces of zlib/libiconv in if they are not explicitly told not to.  Those packages have those options available for a reason, and I'm assuming that reason is to ensure that zlib/libiconv code does not end up in my final application.  I'd prefer to be able to turn those off and allow #ifdef's to make sure that those libs are not in the final mix.

Reply:
Updated